lib/utils/key_value_repository.dart

```diff
@@ -18,7 +18,11 @@ class KeyValueRepository with LogMixin {
 
   /// [value] must have a type that is supported by sembast
   Future<dynamic> put(String key, dynamic value) async {
-    logger.info('Put setting with key "$key" and value "$value"');
+    if (key == 'privateKeyKey') {
+      logger.info('Put setting with key "$key"');
+    } else {
+      logger.info('Put setting with key "$key" and value "$value"');
+    }
```

Comment:
What is the meaning of this special string value that only changes the log message?

Reply:
just for security the private key should not be printed without a good reason and it spams the logs as it is quite big

Reply:
In that case I suggest adding an optional named parameter to the put function which can be used to disable logging the value. Checking for a string constant like this adds a really weird and hard to maintain dependency from `KeyValueRepository` to `IdentityService`.